docs/iris/src/sphinxext/generate_logo.py

```diff
@@ -0,0 +1,441 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+
+"""
+Script to generate the Iris logo in every required format.
+Uses XML ElementTree for SVG file editing.
+"""
+
+from collections import OrderedDict, namedtuple
+from copy import deepcopy
+from io import BytesIO
+from os import environ
+from pathlib import Path
+from re import sub as re_sub
+from xml.etree import ElementTree as ET
+from xml.dom import minidom
+
+from cartopy import crs as ccrs, __version__ as cartopy_version
+from cartopy.feature import LAND
+from matplotlib import pyplot as plt, rcParams
+from numpy import linspace
+
+print("LOGO GENERATION START ...")
+
+################################################################################
+# Configuration
+
+# How much bigger than the globe the iris clip should be.
+CLIP_GLOBE_RATIO = 1.28
+
+# Pixel size of the square logo.
+LOGO_PIXELS = 1024
+
+# Banner width, text and text size must be manually tuned to work together.
+# Pixel dimensions of text banner.
+BANNER_PIXELS = {"width": 588, "height": 256}
+# Text printed in the banner.
+BANNER_TEXT = "Iris"
+# How much smaller than the globe the banner text should be.
+TEXT_GLOBE_RATIO = 0.6
+
+WRITE_DIRECTORY = Path("_static")
+# Start of all filenames to be written.
+FILENAME_PREFIX = environ["PROJECT_PREFIX"]
+
+# The logo's SVG elements can be configured at their point of definition below.
+
+################################################################################
+
+# Establish some sizes and ratios.
+# Set plot to include space for the mask to be added.
+# figure_inches doesn't influence the final size, but does influence the coastline definition.
+figure_inches = 10
+mpl_points_per_inch = 72
+
+globe_inches = figure_inches / CLIP_GLOBE_RATIO
+pad_inches = (figure_inches - globe_inches) / 2
+background_points = figure_inches * mpl_points_per_inch
+globe_points = globe_inches * mpl_points_per_inch
+
+mpl_font_points = rcParams["font.size"]
+inch_font_ratio = mpl_points_per_inch / mpl_font_points
+
+# XML ElementTree setup.
+namespaces = {"svg": "http://www.w3.org/2000/svg"}
+ET.register_namespace("", namespaces["svg"])
+
+################################################################################
+# Create new SVG elements for logo.
+
+# The elements that are displayed (not just referenced).
+# Order is important for layering.
+artwork_dict = OrderedDict.fromkeys(["background", "glow", "sea", "land",])
```

Comment:
Python 2 support was dropped, wasn't it? The normal `dict` would be ordered anyway.

Reply:
It's frankly embarrassing that I didn't already know that `dict`s as now ordered!